File: msgraph-mail/ruby/graphrubyv4/users/attachment.rb
require 'microsoft_kiota_abstractions'
require_relative './entity'
require_relative './users'

module Graphrubyv4::Users
    class Attachment < Graphrubyv4::Users::Entity
        include MicrosoftKiotaAbstractions::Parsable
        ## 
        # The MIME type.
        @content_type
        ## 
        # true if the attachment is an inline attachment; otherwise, false.
        @is_inline
        ## 
        # The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z
        @last_modified_date_time
        ## 
        # The display name of the attachment. This does not need to be the actual file name.
        @name
        ## 
        # The length of the attachment in bytes.
        @size
        ## 
        ## Gets the contentType property value. The MIME type.
        ## @return a string
        ## 
        def  content_type
            return @content_type
        end
        ## 
        ## Gets the isInline property value. true if the attachment is an inline attachment; otherwise, false.
        ## @return a boolean
        ## 
        def  is_inline
            return @is_inline
        end
        ## 
        ## Gets the lastModifiedDateTime property value. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z
        ## @return a date_time_offset
        ## 
        def  last_modified_date_time
            return @last_modified_date_time
        end
        ## 
        ## Gets the name property value. The display name of the attachment. This does not need to be the actual file name.
        ## @return a string
        ## 
        def  name
            return @name
        end
        ## 
        ## Gets the size property value. The length of the attachment in bytes.
        ## @return a integer
        ## 
        def  size
            return @size
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return super.merge({
                "contentType" => lambda {|o, n| o.content_type = n.get_string_value() },
                "isInline" => lambda {|o, n| o.is_inline = n.get_boolean_value() },
                "lastModifiedDateTime" => lambda {|o, n| o.last_modified_date_time = n.get_date_value() },
                "name" => lambda {|o, n| o.name = n.get_string_value() },
                "size" => lambda {|o, n| o.size = n.get_number_value() },
            })
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super
            writer.write_string_value("contentType", @content_type)
            writer.write_boolean_value("isInline", @is_inline)
            writer.write_date_value("lastModifiedDateTime", @last_modified_date_time)
            writer.write_string_value("name", @name)
            writer.write_number_value("size", @size)
        end
        ## 
        ## Sets the contentType property value. The MIME type.
        ## @param value Value to set for the contentType property.
        ## @return a void
        ## 
        def  content_type=(contentType)
            @content_type = contentType
        end
        ## 
        ## Sets the isInline property value. true if the attachment is an inline attachment; otherwise, false.
        ## @param value Value to set for the isInline property.
        ## @return a void
        ## 
        def  is_inline=(isInline)
            @is_inline = isInline
        end
        ## 
        ## Sets the lastModifiedDateTime property value. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z
        ## @param value Value to set for the lastModifiedDateTime property.
        ## @return a void
        ## 
        def  last_modified_date_time=(lastModifiedDateTime)
            @last_modified_date_time = lastModifiedDateTime
        end
        ## 
        ## Sets the name property value. The display name of the attachment. This does not need to be the actual file name.
        ## @param value Value to set for the name property.
        ## @return a void
        ## 
        def  name=(name)
            @name = name
        end
        ## 
        ## Sets the size property value. The length of the attachment in bytes.
        ## @param value Value to set for the size property.
        ## @return a void
        ## 
        def  size=(size)
            @size = size
        end
    end
end

File: msgraph-mail/php/Users/Attachment.php
<?php

namespace Microsoft\Graph\Users;

use Microsoft\Kiota\Abstractions\Serialization\ParseNode;

class Attachment extends Entity 
{
    /** @var string $contentType The MIME type. */
    private string $contentType;
    
    /** @var bool $isInline true if the attachment is an inline attachment; otherwise, false. */
    private bool $isInline;
    
    /** @var DateTimeOffset $lastModifiedDateTime The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z */
    private DateTimeOffset $lastModifiedDateTime;
    
    /** @var string $name The display name of the attachment. This does not need to be the actual file name. */
    private string $name;
    
    /** @var int $size The length of the attachment in bytes. */
    private int $size;
    
    /**
     * Gets the contentType property value. The MIME type.
     * @return string|null
    */
    public function getContentType(): ?string {
        return $this->contentType;
    }

    /**
     * Gets the isInline property value. true if the attachment is an inline attachment; otherwise, false.
     * @return bool|null
    */
    public function getIsInline(): ?bool {
        return $this->isInline;
    }

    /**
     * Gets the lastModifiedDateTime property value. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z
     * @return DateTimeOffset|null
    */
    public function getLastModifiedDateTime(): ?DateTimeOffset {
        return $this->lastModifiedDateTime;
    }

    /**
     * Gets the name property value. The display name of the attachment. This does not need to be the actual file name.
     * @return string|null
    */
    public function getName(): ?string {
        return $this->name;
    }

    /**
     * Gets the size property value. The length of the attachment in bytes.
     * @return int|null
    */
    public function getSize(): ?int {
        return $this->size;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable>
    */
    public function getFieldDeserializers(): array {
        echo('This is the body of the deserializer.');
    }

    /**
     * Serializes information the current object
     * @param ?ISerializationWriter $writer $writer Serialization writer to use to serialize this model
    */
    public function serialize(?ISerializationWriter $writer): void {
        parent::serialize($writer);
        $writer->writeStringValue('contentType', $this->contentType);
        $writer->writeBooleanValue('isInline', $this->isInline);
        $writer->writeObjectValue('lastModifiedDateTime', $this->lastModifiedDateTime);
        $writer->writeStringValue('name', $this->name);
        $writer->writeObjectValue('size', $this->size);
    }

    /**
     * Sets the contentType property value. The MIME type.
     *  @param string|null $value Value to set for the contentType property.
    */
    public function setContentType(string $value): void {
        $this->contentType = $value;
    }

    /**
     * Sets the isInline property value. true if the attachment is an inline attachment; otherwise, false.
     *  @param bool|null $value Value to set for the isInline property.
    */
    public function setIsInline(bool $value): void {
        $this->isInline = $value;
    }

    /**
     * Sets the lastModifiedDateTime property value. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is 2014-01-01T00:00:00Z
     *  @param DateTimeOffset|null $value Value to set for the lastModifiedDateTime property.
    */
    public function setLastModifiedDateTime(DateTimeOffset $value): void {
        $this->lastModifiedDateTime = $value;
    }

    /**
     * Sets the name property value. The display name of the attachment. This does not need to be the actual file name.
     *  @param string|null $value Value to set for the name property.
    */
    public function setName(string $value): void {
        $this->name = $value;
    }

    /**
     * Sets the size property value. The length of the attachment in bytes.
     *  @param int|null $value Value to set for the size property.
    */
    public function setSize(int $value): void {
        $this->size = $value;
    }

}
